tools/mksurfdata_map/src/mkpftMod.F90

```diff
@@ -509,23 +513,22 @@ subroutine mkpft(ldomain, mapfname, fpft, ndiag, allow_no_crops, &
 
      call domain_checksame( tdomain, ldomain, tgridmap )
 
+     ! Obtain mask_src and frac_dst
+     mask_src(:) = tdomain%mask(:)
+     call gridmap_calc_frac_dst(tgridmap, mask_src, frac_dst)
      ! Area-average percent cover on input grid [pctpft_i] to output grid 
      ! [pctpft_o] and correct [pctpft_o] according to land landmask
      ! Note that percent cover is in terms of total grid area.
-  
-     do no = 1,ns_o
-        pctlnd_o(no)     = tgridmap%frac_dst(no) * 100._r8
-        ldomain%frac(no) = tgridmap%frac_dst(no) 
```

Comment:
In reviewing your latest changes, I noticed that you no longer set ldomain%frac in this subroutine. Is that intentional - e.g., did you determine that that doesn't need to be set here? If so, I think ldomain can be changed to an input argument in this subroutine.

Reply:
Changed ldomain to an input argument.